check.go

```diff
@@ -319,6 +314,10 @@ func (c *CheckRunner) UpdateCheck(checkID types.CheckID, status, output string)
 		delete(c.checksCritical, checkID)
 	}
 
+	// Do nothing if update is idempotent
+	if check.Status == status && check.Output == output {
+		return
```

Comment:
Just to be sure I'm understanding things... moving this return is the 'fix'. So it doesn't skip things between the old return and this one? Wanted to be sure as much of this just seems like a refactor/cleanup but it is fixing something and how this fixes it wasn't called out anywhere (or I missed it).

Reply:
1, the bug is , if check.Status is cirtical at first(which esm got from consul-agent), and esm checked it again and found it's cirtical, then this process will return. But in fact we need to do something with it.
2, I had checked that this bug  is since 0350cc6. I just revert the code to before commit 0350cc6.